pkg/pf/tests/logging_test.go

```diff
@@ -0,0 +1,72 @@
+package tfbridgetests
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/pf/tests/internal/providerbuilder"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/pf/tests/pulcheck"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/pf/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/tokens"
+)
+
+func TestLogCaputure(t *testing.T) {
+	originalStdout := os.Stdout
+	r, w, err := os.Pipe()
```

Comment:
I think you also need to verify this stdout because the provider is ran in the test process and not by the engine.

Reply:
I actually just removed the os.Stdout stuff. The test asserts that `RedirectLogs` is hooked up an operating: selectively including logs. There are already tests that `RedirectLogs` elides stuff.

Also, I think that Go's test hardness already redirects `log` output, this doesn't do anything.

Reply:
Good point, I think the assertions are enough indeed